src/Corax/IndexWriter.cs

```diff
@@ -78,21 +81,42 @@ public IndexWriter([NotNull] StorageEnvironment environment, IndexFieldsMapping
             _postingListContainerId = Transaction.OpenContainer(Constants.IndexWriter.PostingListsSlice);
             _entriesContainerId = Transaction.OpenContainer(Constants.IndexWriter.EntriesContainerSlice);
             _jsonOperationContext = JsonOperationContext.ShortTermSingleUse();
+            if (fieldsMapping != null)
+            {
+                fieldsMapping.UpdateMaximumOutputAndTokenSize();
+                _encodingBufferHandler = ArrayPool<byte>.Shared.Rent(fieldsMapping.MaximumOutputSize);
+                _tokensBufferHandler = ArrayPool<Token>.Shared.Rent(fieldsMapping.MaximumTokenSize);
+            }
             _fieldsMapping = fieldsMapping ?? IndexFieldsMapping.Instance;
+            
+            var bufferSize = fieldsMapping?.Count ?? 1024;
+            _buffer = new Dictionary<Slice, List<long>>[bufferSize];
+            for (int i = 0; i < bufferSize; ++i)
+                _buffer[i] = new Dictionary<Slice, List<long>>(SliceComparer.Instance);
```

Comment:
Isn't that quite a lot? Under what scenario we don't have a `fieldMapping` ?
You are also creating a _lot_ of dictionary instances here, under what scenario they will be used?

Reply:
This is a container for entries before writing into the index.

It was made specially for backward compatibility. I've refactored old tests so we don't need it anymore.